src/ncp-spinel/SpinelNCPInstance.cpp

```diff
@@ -694,7 +694,7 @@ SpinelNCPInstance::set_property(
 		} else if (strcaseequal(key.c_str(), kWPANTUNDProperty_NetworkPANID)) {
 			uint16_t panid = any_to_int(value);
 
-			mPanid = panid;
+			mCurrentNetworkInstance.panid = panid;
```

Comment:
This shouldn't be necessary, and doesn't follow the right model.

We set the PANID on the NCP, and then the NCP tells us that it set the PANID by telling us the new value of the PANID. We don't set the PANID or XPANID directly, we let the NCP tell us that these values have changed.

I'm willing to consider changes to that model with a good justification though.